### DISCUSSION TITLE
Living document with all identified enhancements for ferdium-app and other repos

### DISCUSSION THREAD

**Discussion by vraravam** [ferdium/ferdium-app#1295] [Ideas]
1) `adonis` upgrade in both the `internal-server` and the `ferdium-server` codebases
2) Converge the `internal-server` folder within the `ferdium-app`, and the functional equivalent code in the `ferdium-server` codebases. Possible solution is to make them as a single git-submodule, and reference in both codebases.
3) Move recipes from a git submodule to a standalone service that is exposed via http apis. The installed application should then be downloading (after searching) from this standalone service. This will reduce the installer size. But, it will also mean that we need to have a robust mechanism in each recipe to state what versions of Ferdium it is compatible with.
4) Remove deprecations after `mobx` upgrade to v6 (https://mobx.js.org/enabling-decorators.html)
5) Fix tech-debts introduced during the typescript conversion
6) build tooling for all(?) codebases (vite?)
7) Add ability to collapse Workspace drawer into icons. (targetted behavior somewhat like Jira sidebar)
8) Add more tests

**Comment by mcmxcdev** [comment]
All very good points, Vijay!

These come from the top of my head:

- investigate if it makes sense to switch to https://github.com/electron/forge
- [replace deprecated dependencies with modern alternatives](https://github.com/ferdium/ferdium-app/issues/1284)
- stop excessive log spam in dev tools
- implement [eslint-plugin-mobx](https://github.com/mobxjs/mobx/tree/main/packages/eslint-plugin-mobx)
